torchgeo/datasets/splits.py

```diff
@@ -0,0 +1,375 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Dataset splitting utilities."""
+
+from copy import deepcopy
+from math import floor
+from typing import Any, List, Optional, Sequence, Tuple, Union
+
+from rtree.index import Index, Property
+from torch import Generator, default_generator, randint, randperm
+from torch._utils import _accumulate
+from torch.utils.data import Subset, TensorDataset, random_split
+
+from ..datasets import GeoDataset, NonGeoDataset
+from .utils import BoundingBox
+
+__all__ = (
+    "random_nongeo_split",
+    "random_bbox_assignment",
+    "random_bbox_splitting",
+    "random_grid_cell_assignment",
+    "roi_split",
+    "time_series_split",
+)
+
+
+def _fractions_to_lengths(fractions: Sequence[float], total: int) -> Sequence[int]:
+    """Utility to divide a number into a list of integers according to fractions.
+
+    Args:
+        fractions: list of fractions
+        total: total to be divided
+
+    Returns:
+        List of lengths.
+
+    .. versionadded:: 0.4
+    """
+    lengths = [floor(frac * total) for frac in fractions]
+    remainder = int(total - sum(lengths))
+    # add 1 to all the lengths in round-robin fashion until the remainder is 0
+    for i in range(remainder):
+        idx_to_add_at = i % len(lengths)
+        lengths[idx_to_add_at] += 1
+    return lengths
+
+
+def _create_geodataset_like(dataset: GeoDataset, index: Index) -> GeoDataset:
+    """Utility to create a new GeoDataset from an existing one with a different index.
+
+    Args:
+        dataset: dataset to copy
+        index: new index
+
+    Returns:
+        A new GeoDataset.
+
+    .. versionadded:: 0.4
+    """
+    new_dataset = deepcopy(dataset)
+    new_dataset.index = index
+    return new_dataset
+
+
+def random_nongeo_split(
```

Comment:
We might actually be able to get rid of this. `torch.utils.data.random_split` now supports fractions as well, so we no longer need to do anything special, we can use the builtin method.

Reply:
Removed!